packages/framework/src/types/discover.types.ts

```diff
@@ -67,6 +68,13 @@ export type DiscoverWorkflowOutput = {
   };
 };
 
+export type Workflow<T_Payload = any> = {
+  trigger: (
+    event: Prettify<Omit<EventTriggerParams<T_Payload>, 'workflowId' | 'bridgeUrl'>>
+  ) => Promise<EventTriggerResult>;
```

Comment:
I thought about adding [`triggerBulk`](https://docs.novu.co/api-reference/events/bulk-trigger-event) and [`broadcast`](https://docs.novu.co/api-reference/events/broadcast-event-to-all) here.

`broadcast` is straightforward to add.

`triggerBulk` needs some more thinking about error handling. If one of the bulk events has an invalid payload as determined with schema validation, there's a question of how that should be surfaced on the method results. Should it do something like a `Promise.allSettled(...)` and potentially have a mixture of successful and failed triggers, or should we throw a failure state if at least 1 of the payloads is invalid?

Reply:
Raised https://linear.app/novu/issue/NV-4061/add-triggerbulk-and-triggerbroadcast-to-defined-novuframework toa ddress this.